cookingWebsite/yourCookbook/viewRecipe.php

```diff
@@ -0,0 +1,112 @@
+<?php
+include('config/init.php');
+verifyUserCooking();
+$recipeId = $_REQUEST['recipeId'];
+$loginId = $_SESSION['loginId'];
+echo yourCookbookEchoHtmlHeader("New Recipe");
+?>
+<script>
+function addTagExistingRecipe(recipeId){
+    var recipeIngredientName = $('#recipeIngredientName').val();
+    var submitItem = $('#submitItem').val();
+    console.log(recipeId, submitItem, recipeIngredientName)
+    $.post('/ajax/saveAddedtag.php', {recipeIngredientName:recipeIngredientName, submitItem:submitItem, recipeId:recipeId}, function(Data){
+          if (Data == "failure"){
+            alert("Error");
+            return false;
+        }
+        // $('.addTag').html(Data)
+    });
+    alert("The tag was sucessfully added!");
+    return false;
+}
+function addRating(loginId, recipeId){
+    var rating = $('#rating').val();
+    var submitItem = $('#submitItem').val();
+    console.log(recipeId, submitItem, loginId)
+    $.post('/ajax/saveRating.php', {rating:rating, submitItem:submitItem, recipeId:recipeId, loginId:loginId}, function(Data){
```

Comment:
It shouldn't be necessary to pass a login id. The ajax endpoint has access to the session also, so you can just look at it there. Also, if the back end actually looks at this loginId and trusts it, that means anyone could impersonate a different user.